server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1074,35 +1083,29 @@ public void onFailure(String source, Exception e) {
             @Override
             public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
                 if (runningSnapshot == null) {
-                    threadPool.generic().execute(ActionRunnable.wrap(listener, l ->
-                        repositoriesService.repository(repositoryName).getRepositoryData(
-                            ActionListener.wrap(repositoryData -> {
-                                Optional<SnapshotId> matchedEntry = repositoryData.getSnapshotIds()
-                                    .stream()
-                                    .filter(s -> s.getName().equals(snapshotName))
-                                    .findFirst();
-                                // If we can't find the snapshot by the given name in the repository at all or if the snapshot we find in
-                                // the repository is not the one we expected to find when waiting for a finishing snapshot we fail.
-                                if (matchedEntry.isPresent()) {
-                                    deleteCompletedSnapshot(
-                                        new Snapshot(repositoryName, matchedEntry.get()), repositoryData.getGenId(), Priority.NORMAL, l);
-                                } else {
-                                    l.onFailure(new SnapshotMissingException(repositoryName, snapshotName));
-                                }
-                            }, l::onFailure))));
+                    resolveAndDeleteSnapshots(snapshotNames, listener, repositoryName);
                     return;
                 }
                 logger.trace("adding snapshot completion listener to wait for deleted snapshot to finish");
                 addListener(runningSnapshot, ActionListener.wrap(
                     result -> {
                         logger.debug("deleted snapshot completed - deleting files");
-                        deleteCompletedSnapshot(
-                            new Snapshot(repositoryName, result.v2().snapshotId()), result.v1().getGenId(), Priority.IMMEDIATE, listener);
+                        final RepositoryData newRepoData = result.v1();
```

Comment:
Here and in the `abortedDuringInit` case I made it so that the remaining deletes will run after the abort is through if the delete matched both a running and existing snapshots.
There is a bit of weirdness here in that we abort the matching snapshot before we check for existing snapshots in the repo. We will have to discuss what exact mechanics we want for the multi-delete when it partially matches.
The easiest to implement would be to ignore missing snapshots and maybe just either throw a `404` if nothing is matched for a delete or go with it if anything matches.